docs/source/onedal/algorithms/graph/subgraph-isomorphism.rst

```diff
@@ -0,0 +1,100 @@
+.. ******************************************************************************
+.. * Copyright 2021 Intel Corporation
+.. *
+.. * Licensed under the Apache License, Version 2.0 (the "License");
+.. * you may not use this file except in compliance with the License.
+.. * You may obtain a copy of the License at
+.. *
+.. *     http://www.apache.org/licenses/LICENSE-2.0
+.. *
+.. * Unless required by applicable law or agreed to in writing, software
+.. * distributed under the License is distributed on an "AS IS" BASIS,
+.. * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. * See the License for the specific language governing permissions and
+.. * limitations under the License.
+.. *******************************************************************************/
+
+.. default-domain:: cpp
+
+.. _alg_subgraph_isomorphism:
+
+====================
+Subgraph isomorphism
+====================
+
+.. include::  ../../../includes/graph/subgraph-isomorphism-introduction.rst
+
+------------------------
+Mathematical formulation
+------------------------
+
+.. _subgraph_isomorphism_compute:
+
+Subgraphs definition
+--------------------
+
+A graph :math:`H = (V'; E')` is called a subgraph of graph :math:`G = (V; E)` if 
+:math:`V' \subseteq V; E' \subseteq E` and :math:`V'` contains all the endpoints of all the 
+edges in :math:`E'` [Gross2014]_.
+
+Further we denote the :capterm:`induced subgraph on the vertex set<Induced subgraph on the vertex set>` 
+as **induced** subgraph, the :capterm:`induced subgraph on the edge set<Induced subgraph on the edge set>`
+as **non-induced** subgraph.
+
+Computing
+---------
+
+Given two graphs :math:`G` and :math:`H`, the problem is to determine whether graph :math:`G` contains
+a subgraph isomorphic to graph :math:`H` and find the exact mapping of subgraph :math:`H` in
+graph :math:`G`.
+
+:math:`G` is called **target** graph, :math:`H` is called **pattern** graph.
+
+Mapping is a bijection or one-to-one correspondence between vertices of :math:`H` and subgraph of 
+graph :math:`G`. Every existing edge corresponds two vertices adjacency relationship, non-existing edge 
+corresponds two vertices non-adjacency relationship. Induced kind of subgraph isomorphims preserves
+adjacency and non-adjacency relationship between vertices, when non-induced kind preserveds only 
+adjacency relationship.
```

Comment:
maybe like this?

```suggestion
graph :math:`G`. Two vertices are adjacent if there is an existing edge between them, and non-adjacent otherwise.
Induced subgraph isomorphism preserves both adjacency and non-adjacency relationships between vertices,
while non-induced subgraph isomorphism preserves only adjacency relationship.
```

note that you have a typo in isomorphism

Reply:
if you want to have something closer to your original text, then maybe this is also ok:
```suggestion
graph :math:`G`. Each existing edge denotes adjacency relationship between two vertices, while each non-existing edge 
denotes non-adjacency relationship between two vertices. Induced kind of subgraph isomorphism preserves
adjacency and non-adjacency relationship between vertices, while non-induced kind preserves only 
adjacency relationship.
```